core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -132,6 +133,8 @@ static void toJson(Type.PrimitiveType primitive, JsonGenerator generator) throws
   static void toJson(Type type, JsonGenerator generator) throws IOException {
     if (type.isPrimitiveType()) {
       toJson(type.asPrimitiveType(), generator);
+    } else if (type.isVariantType()) {
+      generator.writeString(type.toString());
```

Comment:
Can you update `TestSchemaParser`?

Reply:
Currently we don't have `TestSchemaParser`.  I'm trying to use TestMetadataUpdateParser.java indirectly test the change here. Let me know if we should create `TestSchemaParser`.